.github/workflows/ci-build.yml

```diff
@@ -1,31 +1,31 @@
-# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
-name: Node.js CI
+name: Node.js Build and Test
 
 on:
   push:
     branches:
       - main
   pull_request:
-  
+
 jobs:
   build:
-
     runs-on: ubuntu-latest
     timeout-minutes: 10
-
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x, 20.x]
-
+        node-version: [18.x, 20.x]
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
+      uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm install
-    - run: npm run setup
-    - run: npm run test
-    # - run: npm run coverage
+    - name: Get Development Dependencies
+      run: npm i
+    - name: Build and Run Tests in Each Package
+      run: |
+        for pkg in packages/logger packages/oauth packages/rtm-api packages/socket-mode packages/types packages/web-api packages/webhook; do
```

Comment:
Ooo those are great ideas! I will play around with that.

Reply:
@zimeg take a look at this PR: https://github.com/filmaj/node-slack-sdk/pull/3
It implements your suggestion. The workflow file looks much nicer but what do you think about the 14 status checks now posted to the PR? I don't mind it and at a glance you can see exactly which project is passing/failing.

Reply:
Another nice thing about this approach is that it runs each matrix combination in parallel, so feedback is faster.

Reply:
Nice! Thanks for experimenting with this, I didn't expect that 14 statuses would appear!

It's a bit busier, but I think it's pretty nice that statuses for each package and version are shown separately. The parallel running is sweet too! I'm definitely a fan, but your call on making the change.

Reply:
:two-cents: I think that the 14 statuses provides good visual feedback to figure out what versions and packages are failing, also big fan of parallel jobs since it enables us to fail faster